### PR TITLE
add workers to fNatsSubscriberTransport

### PR DESCRIPTION
**Pull request by kevinfrommelt-wf**
### Story:
Need to add support for parallel workers for processing subscription messages.

### Acceptance Criteria:
- [X] Code has been tested and results documented
- [ ] Unit tests have been updated
- [ ] Updates to documentation if necessary
- [ ] Verify and document changes to any other Messaging components
- [X] Pull request made against the 'develop' branch, not master

### Design Notes:
Added a `FNatsSubscriberFactoryBuilder` similar to `FNatsServerFactoryBuilder`
Worker queue logic is also similar to the nats server worker queue logic

### How To Test:
Modified vessel code examples to process messages too slowly with a single worker. Then updated the subscriber to use multiple workers. 

### My Test Results:
Regardless of worker count all messages were processed. With a higher worker count it was able to process more messages.

#### Reviewers:
@Workiva/service-platform

### CODE REVIEW

File: lib/go/nats_scope_transport.go [hunk 1]
```diff
@@ -139,25 +146,31 @@ func (n *FNatsSubscriberTransportFactory) GetTransport() FSubscriberTransport {
...
 // for pub/sub. Subscribers using this transport will subscribe to the provided
 // queue, forming a queue group. When a queue group is formed, only one member
 // receives the message.
 func NewNatsFSubscriberTransportWithQueue(conn *nats.Conn, queue string) FSubscriberTransport {
-	return &fNatsSubscriberTransport{conn: conn, queue: queue}
+	return &fNatsSubscriberTransport{conn: conn, queue: queue, workerCount: 1, workC: make(chan *nats.Msg, 64)}
+}
+
+func NewNatsFSubscriberTransportWithQueueAndWorker(conn *nats.Conn, queue string, workerCount uint) FSubscriberTransport {
+	return &fNatsSubscriberTransport{conn: conn, queue: queue, workerCount: workerCount, workC: make(chan *nats.Msg, 64)}
```
Comment (kevinfrommelt-wf) [inline lib/go/nats_scope_transport.go:173]: Should the `64`s be replaced by `defaultWorkQueueLength`?

**Comment by aviary-wf** [comment]
## Security Insights
<details>
<summary>(1) Vulnerable direct dependencies were detected</summary>
<li>1 vulns in <code>aiohttp < 3.7.4</code> via <code>lib/python/requirements_dev_asyncio.txt</code></li>
</details>

## Action Items
* Review dependencies for available updates
* See <a href="https://splunk.workiva.net/en-US/app/search/security_vulnerable_dependencies?form.repository=frugal">this Splunk dashboard</a> for more CVE details
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/go/nats_scope_transport.go [hunk 2]
```diff
@@ -111,25 +113,75 @@ func (n *fNatsPublisherTransport) formattedSubject(subject string) string {
 	return fmt.Sprintf("%s%s", frugalPrefix, subject)
 }
 
+// FNatsServerBuilder configures and builds NATS subscribers.
```
Comment (kevinfrommelt-wf) [inline lib/go/nats_scope_transport.go:114]: FNatsSubscriberBuilder

**Review by brianshannan-wf — APPROVED** [review]

**Comment by kevinfrommelt-wf** [comment]
QA +1

**Comment by houstonking-wf** [comment]
@Workiva/release-management-p for merge

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**